e2e/evm_test.go

```diff
@@ -83,356 +101,371 @@ func getTestCases(t *testing.T, protocolStr protocolStr, serviceIDOverride proto
+func setupSIGINTHandler(ctx context.Context, cancel context.CancelFunc, t *testing.T) {
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt)
+	go func() {
+		<-sigCh
+		fmt.Println("🛑 Received SIGINT, cancelling test...")
+		cancel()
+
+		// Give a short time for cleanup to happen in the other handlers
+		// but don't hang forever
+		timer := time.NewTimer(5 * time.Second)
+		select {
+		case <-timer.C:
+			fmt.Println("Cleanup timed out, forcing exit...")
+			os.Exit(1)
+		}
+	}()
+}
+
+// logEVMTestStartInfo logs the test start information for the user.
+func logEVMTestStartInfo(opts testOptions) {
+	fmt.Println("\n🌿 Starting PATH E2E EVM test ...")
 	fmt.Printf("  🧬 Gateway URL: %s\n", opts.gatewayURL)
 	fmt.Printf("  📡 Test protocol: %s\n", opts.testProtocol)
 	if opts.serviceIDOverride != "" {
 		fmt.Printf("  ⛓️  Running tests for service ID: %s\n", opts.serviceIDOverride)
 	} else {
 		fmt.Printf("  ⛓️  Running tests for all service IDs\n")
 	}
+}
 
-	// TODO_NEXT: This arbitrary wait is somewhat hacky and may need to be revisited in the future.
-	//
-	// Wait for several rounds of hydrator checks to complete to ensure invalid endpoints are sanctioned.
-	// 		ie. for returning empty or invalid responses, etc.
+// waitForHydratorIfNeeded waits for several rounds of hydrator checks if configured.
+func waitForHydratorIfNeeded(opts testOptions) {
 	if opts.waitForHydrator > 0 {
-		fmt.Printf("⏰ Waiting for %d seconds before starting tests to allow several rounds of hydrator checks to complete...\n", opts.waitForHydrator)
+		fmt.Printf("\n⏰ Waiting for %d seconds before starting tests to allow several rounds of hydrator checks to complete...\n", opts.waitForHydrator)
 		if os.Getenv("CI") != "" || os.Getenv("GITHUB_ACTIONS") != "" {
-			<-time.After(time.Duration(opts.waitForHydrator) * time.Second) // In CI, do not use wait bar as CI logs do not support it.
+			<-time.After(time.Duration(opts.waitForHydrator) * time.Second)
 		} else {
-			showWaitBar(opts.waitForHydrator) // In local environment, show progress bar to indicate we're waiting.
+			showWaitBar(opts.waitForHydrator)
 		}
 	}
+}
 
-	// Get test cases based on protocol
-	testCases := getTestCases(t, opts.testProtocol, opts.serviceIDOverride)
-
-	// Initialize map to store service summaries
-	serviceSummaries := make(map[protocol.ServiceID]*serviceSummary)
+// runEVMServiceTest runs the E2E test for a single EVM service in a test case.
+func runEVMServiceTest(
+	t *testing.T,
+	ctx context.Context,
+	tc testCase,
+	opts testOptions,
+	summary *serviceSummary,
+) (serviceTestFailed bool) {
+	results := make(map[jsonrpc.Method]*methodMetrics)
+	var resultsMutex sync.Mutex
+
+	fmt.Printf("\n🛠️  Running EVM test: %s%s%s\n\n", BOLD_BLUE, tc.name, RESET)
+
+	// Validate that all methods have a definition
+	for _, method := range tc.methods {
+		if _, exists := tc.methodConfigs[method]; !exists {
+			t.Fatalf("No definition for method %s", method)
```

Comment:
As suggested above, dropping the `methods` field and doing a range for loop over `methodConfigs`  removes the need for this check.